### PR TITLE
CAS-8048 MSSummary refactor for performance increase

### PR DESCRIPTION
**Pull request by dmehring**
This refactor, focused on MSSummary, significantly increases performance and simplifies maintainability.

### CODE REVIEW

**Comment by dmehring** [comment]
@gervandiepen @juliantaylor @tammojan Would someone be kind enough to review and merge these changes. I submitted this PR over a week ago. Thanks.

File: ms/MSOper/MSMetaData.cc [hunk 1]
```diff
@@ -1994,16 +1994,31 @@ SHARED_PTR<Vector<Double> > MSMetaData::_getTimes() const {
...
+
+	/*
 	String colName = MeasurementSet::columnName(MSMainEnums::EXPOSURE);
 	ScalarColumn<Double> exposure (*_ms, colName);
 	String unit = *exposure.keywordSet().asArrayString("QuantumUnits").begin();
-	SHARED_PTR<QVD > ex(
+	SHARED_PTR<QVD> ex(
 		new QVD(exposure.getColumn(), unit)
 	);
+	*/
```
Comment (tammojan) [inline ms/MSOper/MSMetaData.cc]: Is there a reason for commenting this out instead of deleting it?

File: ms/MSOper/MSMetaData.h [hunk 2]
```diff
@@ -623,6 +630,8 @@ class MSMetaData {
 
 	mutable map<SourceKey, SourceProperties> _sourceInfo;
 
+	//mutable map<SubScanKey, Quantity> _meanExposureTimeMap;
```
Comment (tammojan) [inline ms/MSOper/MSMetaData.h]: Why are you adding a commented piece of code? Is this a todo?

File: ms/MSOper/MSSummary.cc [hunk 3]
```diff
@@ -320,419 +320,164 @@ void MSSummary::listMain (LogIO& os, Record& outRec, Bool verbose,
...
-			++recLength;
-		}
-
-		// post to logger
-		if (verbose) os << LogIO::POST;
-
-		// push OBS/ARR iteration
-		obsarriter.next();
-	} // end of OBS/ARR iteration
+    ArrayKey aryKey;
```
Comment (tammojan) [inline ms/MSOper/MSSummary.cc]: Please keep the tab/space formatting the same within a file. Somehow your editor indented with spaces here, while the rest of the file is indented with tabs.

**Comment by tammojan** [comment]
Apart from the style issues, the changes are fine with me.

I tested with the lofar tree (particularly our msoverview which relies heavily on mssummary) and all went fine.

**Comment by dmehring** [comment]
@tammojan updated as per your comments. Thanks.

**Comment by tammojan** [comment]
Thanks @dmehring . I've squashed your commits and merged this PR to the master manually.

**Pull request closed without merging**